tests/unit/sceneUpdateTests.cpp

```diff
@@ -12,147 +12,160 @@ using namespace Tangram;
-    updates.push_back({"lights.light2.ambient", "0.0"});
-
-    // Delete lights
-    updates.push_back({"lights", "null"});
-    updates.push_back({"lights.light2.ambient", "0.0"});
-
-    // Tangram apply scene updates, reload the scene
+    // Add an update.
+    std::vector<SceneUpdate> updates = {{"map.a", "new_value"}};
+    // Apply scene updates, reload scene.
     SceneLoader::applyUpdates(scene, updates);
-
     const Node& root = scene.config();
-
-    REQUIRE(!root["lights"]["light1"]);
-    REQUIRE(!root["lights"]["light2"]);
+    REQUIRE(root["map"]["a"].Scalar() == "new_value");
 }
 
-TEST_CASE("Scene update tests, multiple updates with globals") {
-    std::shared_ptr<Scene> scene = std::make_shared<Scene>();
-    std::vector<SceneUpdate> updates;
-
-    REQUIRE(!sceneString.empty());
-
-    REQUIRE(SceneLoader::loadConfig(sceneString, scene->config()));
+TEST_CASE("Apply scene update to a nested map entry") {
+    // Setup.
+    Scene scene;
+    REQUIRE(SceneLoader::loadConfig(sceneString, scene.config()));
+    // Add an update.
+    std::vector<SceneUpdate> updates = {{"nest.map.a", "new_value"}};
+    // Apply scene updates, reload scene.
+    SceneLoader::applyUpdates(scene, updates);
+    const Node& root = scene.config();
+    REQUIRE(root["nest"]["map"]["a"].Scalar() == "new_value");
+}
 
-    Node& root = scene->config();
+TEST_CASE("Apply scene update to a sequence node") {
+    // Setup.
+    Scene scene;
+    REQUIRE(SceneLoader::loadConfig(sceneString, scene.config()));
+    // Add an update.
+    std::vector<SceneUpdate> updates = {{"seq", "new_value"}};
+    // Apply scene updates, reload scene.
+    SceneLoader::applyUpdates(scene, updates);
+    const Node& root = scene.config();
+    REQUIRE(root["seq"].Scalar() == "new_value");
+}
 
-    SceneLoader::parseGlobals(root["global"], scene);
-    SceneLoader::applyGlobalRefUpdates(root, scene);
-    SceneLoader::applyGlobalProperties(root, scene);
+TEST_CASE("Apply scene update to a nested sequence node") {
+    // Setup.
+    Scene scene;
+    REQUIRE(SceneLoader::loadConfig(sceneString, scene.config()));
+    // Add an update.
+    std::vector<SceneUpdate> updates = {{"nest.seq", "new_value"}};
+    // Apply scene updates, reload scene.
+    SceneLoader::applyUpdates(scene, updates);
+    const Node& root = scene.config();
+    REQUIRE(root["nest"]["seq"].Scalar() == "new_value");
+}
 
-    REQUIRE(root["cameras"]["iso-camera"]["active"].Scalar() == "true");
-    REQUIRE(root["cameras"]["perspective-camera"]["active"].Scalar() == "false");
+TEST_CASE("Apply scene update to a new map entry") {
+    // Setup.
+    Scene scene;
+    REQUIRE(SceneLoader::loadConfig(sceneString, scene.config()));
+    // Add an update.
+    std::vector<SceneUpdate> updates = {{"map.c", "new_value"}};
+    // Apply scene updates, reload scene.
+    SceneLoader::applyUpdates(scene, updates);
+    const Node& root = scene.config();
+    REQUIRE(root["map"]["c"].Scalar() == "new_value");
```

Comment:
I agree that we would ideally ensure that only the nodes we expect to change are actually changed, but that's a bit of a problem. What do we check? One node? Which one? Why only that one? All other nodes? This is a general problem with testing operations on yaml documents. One solution I've considered is string-ifying the modified document and comparing it to the output yaml you would expect, but there is no one "correct" way to write a yaml structure so it's a meaningless effort. 

There is an infinity of arbitrary truths that we could choose to evaluate for any of these functions. Since our tests can be read as a form of documentation, I'd prefer to keep them logically focused.